fuel-merkle/src/binary/in_memory.rs

```diff
@@ -62,6 +67,72 @@ impl Default for MerkleTree {
     }
 }
 
+#[derive(Default, Debug, Clone)]
+pub struct MerkleRootCalculator {
+    stack: Vec<Node>,
+}
+
+impl MerkleRootCalculator {
+    pub fn new() -> Self {
+        Self { stack: Vec::new() }
+    }
+
+    pub fn push(&mut self, data: &[u8]) {
+        let node = Node::create_leaf(0, data);
+        self.stack.push(node);
+
+        if let Some(top_node) = self.stack.last() {
+            if self.stack.len() > 1 {
```

Comment:
```suggestion\
            if self.stack.len() > 1 {
```

It already should be enough=)